docs/running-on-yarn.md

```diff
@@ -196,6 +196,15 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   It should be no larger than the global number of max attempts in the YARN configuration.
   </td>
 </tr>
+<tr>
+  <td><code>spark.yarn.submit.waitAppCompletion</code></td>
```

Comment:
Yeah the intention was for backwards compatibility.  As you say if they are relying on the current behavior in other scripting it would now break.